src/sentry_plugins/amazon_sqs/plugin.py

```diff
@@ -118,57 +133,76 @@ def forward_event(self, event, payload):
             logger.info("sentry_plugins.amazon_sqs.skip_unconfigured", extra=logging_params)
             return
 
-        # TODO(dcramer): Amazon doesnt support payloads larger than 256kb
-        # We could support this by simply trimming it and allowing upload
-        # to S3
-        message = json.dumps(payload)
-        if len(message) > 256 * 1024:
-            logger.info("sentry_plugins.amazon_sqs.skip_oversized", extra=logging_params)
-            return False
+        boto3_args = {
+            "aws_access_key_id": access_key,
+            "aws_secret_access_key": secret_key,
+            "region_name": region,
+        }
 
-        try:
-            client = boto3.client(
-                service_name="sqs",
-                aws_access_key_id=access_key,
-                aws_secret_access_key=secret_key,
-                region_name=region,
+        def log_and_increment(metrics_name):
+            logger.info(
+                metrics_name, extra=logging_params,
+            )
+            metrics.incr(
+                metrics_name, tags=metric_tags,
             )
 
-            message = {"QueueUrl": queue_url, "MessageBody": message}
+        def s3_put_object(*args, **kwargs):
+            s3_client = boto3.client(service_name="s3", **boto3_args)
+            return s3_client.put_object(*args, **kwargs)
+
+        def sqs_send_message(message):
+            client = boto3.client(service_name="sqs", **boto3_args)
+            send_message_args = {"QueueUrl": queue_url, "MessageBody": message}
             # need a MessageGroupId for FIFO queues
             # note that if MessageGroupId is specified for non-FIFO, this will fail
             if message_group_id:
                 from uuid import uuid4
 
-                message["MessageGroupId"] = message_group_id
+                send_message_args["MessageGroupId"] = message_group_id
                 # if content based de-duplication is not enabled, we need to provide a
                 # MessageDeduplicationId
-                message["MessageDeduplicationId"] = uuid4().hex
+                send_message_args["MessageDeduplicationId"] = uuid4().hex
+            return client.send_message(**send_message_args)
+
+        # wrap S3 put_object and and SQS send message in one try/except
+        s3_bucket = self.get_option("s3_bucket", event.project)
+        try:
+            # if we have an S3 bucket, upload to S3
+            if s3_bucket:
```

Comment:
Should it be `if s3_bucket and len(message) > 256 * 1024: `? Do we want to send through SQS and default to S3 if the message is too large?

Reply:
@NisanthanNanthakumar If there is no S3 bucket specified we still have to quit if the size is larger than 256KB. If the user specifies an S3 bucket, we will just always use it no matter what in order to it more uniform.

Reply:
@scefali ah but now that kinda makes it look like an S3 integration.

But if this is the way we want to move forward, maybe we should change the wording on the form, bc it currently worded like we send small messages in SQS and post oversized messages in S3.

Reply:
@NisanthanNanthakumar Good point, let me bring that up with Phillip